pennylane/devices/qubit/simulate.py

```diff
@@ -296,6 +323,12 @@ def simulate(
 
     has_mcm = any(isinstance(op, MidMeasureMP) for op in circuit.operations)
     if circuit.shots and has_mcm:
+        if execution_kwargs.get("mcm_method", None) == "tree-traversal":
+            n_mcms = sum(isinstance(op, MidMeasureMP) for op in circuit.operations)
+            if 2 * n_mcms + 100 > sys.getrecursionlimit():
+                sys.setrecursionlimit(2 * n_mcms + 100)
```

Comment:
`python -c "import sys; print(sys.getrecursionlimit()); sys.setrecursionlimit(2000); exit()"` will always yield `1000`. I'll reset the value before exiting `simulate` as suggested by @albi3ro . The example in the [release notes](https://pennylane.ai/blog/2024/03/pennylane-release-0.35/) would not work without this change, so I thought it would be best to change it under the hood.

Reply:
Thanks! I'm okay with this. Resolving.